engine/src/main/java/org/terasology/physics/shapes/BoxShape.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.physics.shapes;
+
+import org.terasology.math.geom.Vector3f;
+
+/**
+ * Represents a box shaped collision entity in the physics engine.
+ */
+public interface BoxShape extends CollisionShape {
+    /**
+     * Returns the half extents of the box shape without adding a margin.
```

Comment:
As per the goals of this PR, `BoxShape` currently serves as a wrapper to the `com.bulletphysics.collision.shapes.BoxShape` class in TeraBullet. `getHalfExtentsWithoutMargin` is simply a wrapper around the `com.bulletphysics.collision.shapes.BoxShape.getHalfExtentsWithoutMargin` class. However, I do agree that changing this to `getExtents` will simplify the API. I am not quite sure whether to include this change in the current PR or take it up separately like `Transform` as another PR. I have few other tweaks to the `CollisionShape` family of interfaces in mind too, but I think it would be best to take them up in a separate PR.

Reply:
Ok, just make sure to write it somewhere!